providers/implementations/exchange/dh_exch.c

```diff
@@ -388,11 +393,14 @@ static int dh_set_ctx_params(void *vpdhctx, const OSSL_PARAM params[])
     }
 
     p = OSSL_PARAM_locate_const(params, OSSL_KDF_PARAM_CEK_ALG);
-    if (p != NULL) {
+    if (p != NULL && p->data != NULL && p->data_size != 0) {
```

Comment:
this if needs to be split the same way how the kdf_ukm is handled so the free and NULL assignment is done if `p != NULL` and the rest is done only if `p->data != NULL && p->data_size != 0`